sql/binlog.cc

```diff
@@ -8452,7 +8452,7 @@ TC_LOG::enum_result MYSQL_BIN_LOG::commit(THD *thd, bool all) {
       gtid_state->update_on_rollback(thd);
       thd_get_cache_mngr(thd)->reset();
       // Reset the thread OK status before changing the outcome.
-      if (thd->get_stmt_da()->is_ok())
+      if (!wsrep_must_replay(thd) && thd->get_stmt_da()->is_ok())
```

Comment:
1. At this point, if thd already has any error set, then calling `my_error()` will make the DA will make the assert `!set` in `DA::set_error_status()`. Is it expected that there shall be no error when we reach here?

2. This will cause the thread to not reset the DA if the transaction must be replayed. So, if we are replaying the transaction, then from where does the DA get cleared before the replay?

Reply:
1. The callstack here is
```
MYSQL_BIN_LOG::commit(THD*, bool)
ha_commit_trans(THD*, bool, bool)
trans_commit_stmt(THD*, bool)
mysql_execute_command(THD*, bool)
mysql_parse(THD*, Parser_state*, bool)
```

DA allows overwriting the result (I've put comment in details in the code)
 
2. Please take a looka at Wsrep_replayer_service constructor and destructor and m_da_shadow in wsrep_high_priority_service.cc

3. Commit message first line is Jira issue summary

Reply:
done